Add instructions for manually granting calendar access in README.md

nielthiart opened the pull request:
This adds instructions for adding Claude.app to the TCC database with Calendar access.

If you want to remove the link, you're welcome to copy the content to your docs.

nielthiart committed:
Add instructions for manually granting calendar access in README.md

Omar-V2 commented:
Thanks for adding, this is cool indeed - would you mind adding an additional stronger warning message along the lines of "Only run this if you feel comfortable, know what you are doing..." sort of thing

nielthiart committed:
docs: Add a strong warning about editing TCC database

This update adds a stern warning, more details about editing TCC, a new backup step, and a method to reverse the change without restoring from backup.

nielthiart commented:
> Thanks for adding, this is cool indeed - would you mind adding an additional stronger warning message along the lines of "Only run this if you feel comfortable, know what you are doing..." sort of thing

Good idea. I [added](https://github.com/Omar-V2/mcp-ical/pull/4/commits/51f9ec3a6bf62588da8b0450f029b23d778ef572):

1. A stronger warning.
2. More implementation details.
3. A TCC database backup step.
4. A method to reverse the TCC change without restoring from backup.

This adds quite a bit of content to your README, which is otherwise really straightforward. If you prefer that I move the TCC instruction to a footnote or to a different document, please let me know.

Thanks for considering the PR!

Omar-V2 commented:
Thanks for this, In the interest of keeping the current README straightforward and simple, I think it would be good to move this content to separate location. I'm thinking we can add a docs folder in the root of this repo and call this something like install.md. Keep a one/two liner (with the warnings) in this readme and a link to the new location

Shane-Jay commented:
Hi @nielthiart ,

I implemented a script based on your instructions for granting calendar access via the TCC database. While the script successfully adds the permission, I'm experiencing an issue where MCP can only access the calendar on the first launch of my app (Chatwise). For subsequent calendar access attempts, MCP reports no permission until I restart the app.

Environment details:
- macOS 15.4.1 (also tested on 15.4 with the same issue)
- Using MCP to access calendar data in Chatwise app

The error I'm getting after the first calendar access:
{"content":[{"type":"text","text":"failed to run tool "list_events", mcp server "mcp-ical" was not connected"}],"isError":true}


Here's the script I'm using to grant permissions:
```bash
#!/bin/bash

# Check if Terminal has Full Disk Access permission
if ! ls ~/Library/Application\ Support/com.apple.TCC/TCC.db >/dev/null 2>&1; then
  echo "Error: Terminal does not have Full Disk Access permission. Please enable it in System Settings > Privacy & Security > Full Disk Access."
  exit 1
fi

# Backup TCC database
cp ~/Library/Application\ Support/com.apple.TCC/TCC.db ~/Desktop/TCC_backup.db
if [ $? -eq 0 ]; then
  echo "TCC database has been backed up to ~/Desktop/TCC_backup.db"
else
  echo "Error: Backup failed, please check permissions."
  exit 1
fi

# Grant calendar access permission to Chatwise
sqlite3 ~/Library/Application\ Support/com.apple.TCC/TCC.db <<EOF
REPLACE INTO access 
VALUES(
  'kTCCServiceCalendar',
  'app.chatwise',
  0,
  2,
  0,
  1,
  NULL,
  NULL,
  NULL,
  'UNUSED',
  NULL,
  NULL,
  CAST(strftime('%s','now') AS INTEGER),
  NULL,
  NULL,
  'UNUSED',
  CAST(strftime('%s','now') AS INTEGER)
);
EOF

if [ $? -eq 0 ]; then
  echo "Calendar permission has been granted to app.chatwise"
else
  echo "Error: Failed to grant permission, please check SQLite command or table structure."
  exit 1
fi

Is there something I'm missing in the TCC configuration that would help maintain the connection between MCP and the calendar service throughout the app's lifecycle?

Any insights or suggestions would be greatly appreciated!

nielthiart committed:
docs: update repository URL

nielthiart committed:
docs: Move manual calendar access instructions to docs/install.md

As requested by Omar-V2, this moves the detailed TCC database modification instructions to a separate docs/install.md file and keeps only a brief warning with a link in the README.

nielthiart committed:
docs: formatting and clarity in README.md

- Nest content under list items
- Standardize headings
- Add languages to all code blocks
- LTeX lint language

nielthiart commented:
> Thanks for this, In the interest of keeping the current README straightforward and simple, I think it would be good to move this content to separate location. I'm thinking we can add a docs folder in the root of this repo and call this something like install.md. Keep a one/two liner (with the warnings) in this readme and a link to the new location

@Omar-V2 Agreed. I created a new `docs/install.md` file with detailed instructions (for both methods). I kept the safe method in `README.md`. Since this touches docs, I made minor formatting changes in README.md and ran it through a language linter.

---

@Shane-Jay I suggest we continue this in a [new GitHub issue](https://github.com/Omar-V2/mcp-ical/issues/new), if you think it is a permission issue.

I can't be sure, but based on the log output you shared (`"failed to run tool "list_events", mcp server "mcp-ical" was not connected"`), I think your issue may be between your MCP host (Chatwise) and the MCP server (mcp-ical), unrelated to Calendar permissions for Chatwise.

You can check permissions in System Settings > Privacy & Security to confirm whether your TCC.db change is persisted.

![CleanShot 2025-04-18 at 10 12 08](https://github.com/user-attachments/assets/4b0cab36-6104-4d61-b0a3-03c69792ee54)

Omar-V2 commented:
Merged! Thanks @nielthiart for your contribution!
